### PR TITLE
Add setup.py for setuptools

### PR DESCRIPTION
**Pull request by ObserverOfTime**
This is related to issue #207.

I tried to keep changes to the codebase minimal but I had to edit `syncplay.utils.findWorkingDir` to find the path when installed via `setuptools` since `resources/` is copied into `syncplay/`.
(You could probably use [pkg_resources](https://setuptools.readthedocs.io/en/latest/pkg_resources.html) to load resources and avoid all the hassle.)

Once Syncplay is deployed to [PyPI](https://packaging.python.org/tutorials/packaging-projects/#uploading-the-distribution-archives) (which can be automated by [Travis](https://docs.travis-ci.com/user/deployment/pypi/)), installing it along with its optional dependencies will be as simple as `pip3 install syncplay[gui,tls]`.

**Environment info:**
* OS: `Antergos Linux 5.0.3-arch1-1-ARCH`
* Python: `Python 3.7.2`
* pip: `pip 18.1`
* setuptools: `setuptools 40.8.0`

### CODE REVIEW

**Comment by albertosottile** [comment]
Thank you for this PR, we really appreciate it. We are currently working on implementing `setuptools` for Syncplay [here](https://github.com/albertosottile/syncplay/commits/setuptools), but I see some differences between this PR and our approach that are definitely worthy of a discussion.

Firstly, you used `requirements*.txt` files in the `setup.py` script. That's something we could do, too. 

Secondly, you found a way to install the scripts without restructuring the package, by subclassing the `InstallScripts` class. While we considered that approach, we think that using entry points is the most natural and future-proof choice, given that they are recommended. So, we restructured some of the code to use entry points, instead of install scripts, as you can see [here](https://github.com/albertosottile/syncplay/commit/6f361ff0d10f5c163af9415584e0e314453b3bc5). 

Thirdly, you subclassed `BuildPy` instead of actually moving the `resources` folder inside the `syncplay` one. While I appreciate that this script does not require to relocate the folder, I am afraid for the future support of this solution and of the possible bugs that it could cause. As an example, this `setup.py` script does not currently copy the `resources` folder when running it with `sdist`. Of course, this and other future bugs can be fixed, but maybe it would just be better to move the `resources` folder now and migrate Syncplay to the recommended file structure for a python package. This is the approach that I took in the fork so far.

In summary, I think we will reject this PR in favor of opening a new one, based on the work done so far in the fork linked above. I sincerely apologize for the time you already invested in working on this. In any case, I strongly encourage to review and contribute to the upcoming PR, since you seem very experienced with `setuptools`.

**Comment by ObserverOfTime** [comment]
Thank you for the response. That's all right, I just did it in an evening as a proof of concept mostly. I wasn't aware that you were already working on it in a fork.

**Comment by Et0h** [comment]
Closed as setuptools is now supported via #231
